app/controllers/scim_rails/scim_users_controller.rb

```diff
@@ -129,5 +122,104 @@ def delete
     def get_multi_value_attrs(operation)
       schema_hash = contains_square_brackets?(operation["path"]) ? multi_attr_type_to_value(process_filter_path(operation)) : {}
     end
+
+    def check_allowed_parameters!
+      schema = ScimRails.config.user_schema.dup
+      if schema.fetch(:schemas, []).include?("urn:ietf:params:scim:schemas:core:2.0:User")
+        schema.delete(:schemas)
+        schema.merge!(ParameterService::SCIM_CORE_USER_SCHEMA)
+      end
+
+      bad_fields = ParameterService.invalid_parameters(schema, params)
+      return if bad_fields.empty?
+
+      json_response(
+        {
+          schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+          detail: "Unknown fields: #{bad_fields.join(", ")}",
+          status: "422"
+        },
+        :unprocessable_entity,
+      )
+    end
+
+    def permitted_user_params
+      permitted_params(params, "User").merge(multi_attr_type_to_value(params))
+    end
+
+    def find_value_for(attribute)
+      params.dig(*path_for(attribute))
+    end
+
+    # `path_for` is a recursive method used to find the "path" for
+    # `.dig` to take when looking for a given attribute in the
+    # params.
+    #
+    # Example: `path_for(:name)` should return an array that looks
+    # like [:names, 0, :givenName]. `.dig` can then use that path
+    # against the params to translate the :name attribute to "John".
+
+    def path_for(attribute, object = ScimRails.config.mutable_user_attributes_schema, path = [])
```

Comment:
🌵 I am not completely familiar with this code. But just want to know if we can move this `path_for` method in a model?